### PR TITLE
Created is_palindrome goody

### LINKED ISSUE #233: Add Python `is_palindrome` goody
Would be cool if it works on both strings and lists!

### PR DESCRIPTION
**Pull request by VolodymyrSemenov**
Closes #233.

### CODE REVIEW

File: workspaces/adventure-pack/goodies/python3/src/is_palindrome/__init__.py [hunk 1]
```diff
@@ -0,0 +1,9 @@
+def is_palindrome(arr: list | str) -> bool:
```
Comment (yulianka) [inline workspaces/adventure-pack/goodies/python3/src/is_palindrome/__init__.py:1]: One disadvantage of a name like `arr` is that it suggests a list variable. However, as we are supporting string as well (thank you!) consider a name like `sequence` that is more generalized.

**Review by yulianka — COMMENTED** [review]
Thank you for contributing a Python goody!
To pass the checks you need to do 2 things:
- format the code according to the standard, you can do it by executing `yarn goodies:python3:format` from the `workspaces/adventure-pack` directory.
- re-create snapshots for the website with the command `yarn package-goodies:test -u`.

(`yarn` comes with node, if you don't have this command, README from the adventure pack explains how to get it.)

File: workspaces/adventure-pack/goodies/python3/src/is_palindrome/__init__.py (hunk 1 above)
Comment (VolodymyrSemenov) [reply workspaces/adventure-pack/goodies/python3/src/is_palindrome/__init__.py:1]: Yeah, I just named it arr cause couldn't figure out what to name it, and strings are kinda like arrays. Ill update it now.

Comment (VolodymyrSemenov) [reply workspaces/adventure-pack/goodies/python3/src/is_palindrome/__init__.py]: Should pass the tests now as well. I ended up making a lot of commits, but not sure if that's a problem or how to merge them.

Comment (miorel) [reply workspaces/adventure-pack/goodies/python3/src/is_palindrome/__init__.py]: It's not a problem, they will be squashed into a single commit when the PR is merged 🙂

File: workspaces/adventure-pack/package.json [hunk 2]
```diff
@@ -59,6 +59,7 @@
     "immutability-helper": "3.1.1",
     "invariant": "2.2.4",
     "nullthrows": "1.1.1",
+    "prettier": "^3.3.3",
```
Comment (yulianka) [inline workspaces/adventure-pack/package.json:62]: Please remove `prettier`, as it's already in `devDependencies`. `dependencies` contains packages needed for running and `devDependencies` for developing.

File: workspaces/adventure-pack/goodies/python3/src/is_palindrome/test.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+def test_even_fail():
+    assert not is_palindrome("ABCBBA")
+
+
+def test_odd_fail():
+    assert not is_palindrome("1234221")
+
+
+def test_pass():
```
Comment (miorel) [inline workspaces/adventure-pack/goodies/python3/src/is_palindrome/test.py:12]: I like that we have explicit even and odd fails, could we have the same for pass?

File: workspaces/adventure-pack/goodies/python3/src/is_palindrome/test.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    assert is_palindrome("palindromemordnilap")
+
+
+def test_arr():
```
Comment (miorel) [inline workspaces/adventure-pack/goodies/python3/src/is_palindrome/test.py:16]: More Pythonic to call it list?

```suggestion
def test_list():
```

File: workspaces/adventure-pack/goodies/python3/src/is_palindrome/test.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    assert not is_palindrome([1, 1, 2, 3, 5, 2, 1, 1])
+
+
+def test_single_char():
```
Comment (miorel) [inline workspaces/adventure-pack/goodies/python3/src/is_palindrome/test.py:20]: Rename to "element", then we can have a test for a list as well?

```suggestion
def test_single_element():
```

File: workspaces/adventure-pack/goodies/python3/src/is_palindrome/test.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    assert is_palindrome("#")
+    assert is_palindrome("a")
```
Comment (miorel) [inline workspaces/adventure-pack/goodies/python3/src/is_palindrome/test.py:22]: One more test idea: empty list and empty string!

File: workspaces/adventure-pack/package.json (hunk 2 above)
Comment (VolodymyrSemenov) [reply workspaces/adventure-pack/package.json:62]: It did it automatically :/

Comment (yulianka) [reply workspaces/adventure-pack/package.json:62]: That's strange, it shouldn't happen. Let's revert anyway.

File: workspaces/adventure-pack/goodies/python3/src/is_palindrome/test.py (hunk 6 above)
Comment (VolodymyrSemenov) [reply workspaces/adventure-pack/goodies/python3/src/is_palindrome/test.py:26]: Done

File: workspaces/adventure-pack/goodies/python3/src/is_palindrome/test.py (hunk 5 above)
Comment (VolodymyrSemenov) [reply workspaces/adventure-pack/goodies/python3/src/is_palindrome/test.py]: Done

**Review by yulianka — APPROVED** [review]
Looks good!
